windows/src/engine/kmcomapi/com/keyboards/keymankeyboardsinstalled.pas

```diff
@@ -182,6 +187,46 @@ procedure TKeymanKeyboardsInstalled.Apply;
   end;
 
   Context.Control.AutoApplyKeyman;
+
+  TriggerWindowsLanguageSync;
+end;
+
+// This triggers a synchronisation of the language settings to the
+// cloud, which avoids an issue where language setting changes are
+// overwritten on a subsequent reboot of the system.
+//
+// See #4447 and kmrefresh.cpp for additional detail.
+//
+// Although the sync function is not supported on Windows 7, 8.1,
+// we'll call kmrefresh.exe anyway, as we cannot be sure which version
+// of Windows we are running on -- the calling process may be lied to
+// if it does not have an appropriate manifest. kmrefresh.exe is safe
+// to run as it will simply exit if it does not find the appropriate
+// endpoint to call.
```

Comment:
Does it quietly exit when Windows is offline too?

Reply:
Yes. The sync still needs to run even if Windows is offline and sync is disabled, because it syncs to a local cache which is what is used to overwrite the user settings on next boot (even if sync is disabled!)